File: Data Structures & Algorithms/remove-element/submission-5.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int swapIndex = nums.length;
        int runnerIndex = 0;
        
        while(runnerIndex < swapIndex){
            if (nums[runnerIndex] == val){
                nums[runnerIndex] = nums[--swapIndex];
            } else {
                runnerIndex++;
            }
        }

        return swapIndex;
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-5.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        sol = [1] * len(nums)

        left_product = 1 
        for i in range(len(nums)):
            sol[i] *= left_product
            left_product *= nums[i]
        
        right_product = 1
        for i in range(len(nums)-1, -1, -1):
            sol[i] *= right_product
            right_product *= nums[i]
        
        return sol
